src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -1937,7 +1937,181 @@
+            logging.error(f"Custom LLM streaming error: {e}")
+            yield f"Error: {e}"
+
+    def _openai_stream(self, prompt: str, **kwargs) -> Generator[str, None, None]:
+        """Handle streaming for OpenAI client."""
+        try:
+            # Use the new _build_messages helper method
+            messages, original_prompt = self._build_messages(prompt, kwargs.get('temperature', 0.2), 
+                                                           kwargs.get('output_json'), kwargs.get('output_pydantic'))
+            
+            # Store chat history length for potential rollback
+            chat_history_length = len(self.chat_history)
```

Ruff (0.12.2)

Local variable `chat_history_length` is assigned to but never used

Remove assignment to unused variable `chat_history_length`

(F841)